CRM/MembershipExtras/Page/EditContributionRecurLineItems.php

```diff
@@ -182,25 +182,23 @@ public function run() {
     $this->assign('recurringContribution', $this->contribRecur);
     $this->assign('recurringContributionID', $this->contribRecur['id']);
 
-    $this->assign('periodStartDate', CRM_Utils_Array::value('start_date', $this->contribRecur));
-    $this->assign('periodEndDate', CRM_Utils_Array::value('end_date', $this->contribRecur));
-
     $currentPeriodLineItems = $this->getCurrentPeriodLineItems();
     $this->setCurrentLineItemMembershipTypes($currentPeriodLineItems);
 
     $nextPeriodLineItems = $this->getNextPeriodLineItems();
     $this->setNextLineItemMembershipTypes($nextPeriodLineItems);
 
+    $this->assign('periodStartDate', $this->getEarliestMembershipStartDate($currentPeriodLineItems));
```

Comment:
since $currentPeriodLineItems is now passed a lot, maybe you can move it to class property instead